overlord/snapstate/snapstate.go

```diff
@@ -720,6 +729,12 @@ func Update(st *state.State, name, channel string, revision snap.Revision, userI
 		return nil, err
 	}
 	if len(tts) == 0 && len(updates) == 0 {
```

Comment:
Thank you! You are right of course, this will fail if we only have alias changes. How abou this:
- create new `switch-snap-channel` task
- add this task to `tts` `if len(updates) == 0`

We support channels only for single snap refreshes so we don't need to make doUpdate() itself do more (its doing quite a bit already). If we have an update `len(updates) != 0` the channel will be switched anyway so no need for the extra task. The alias tts changes are then irrelevant and the code does not need to consider them. WDYT?